service/src/main/java/uk/gov/hmcts/reform/fpl/service/casesubmission/CaseSubmissionTemplateDataGenerationService.java

```diff
@@ -0,0 +1,741 @@
+import static uk.gov.hmcts.reform.fpl.enums.YesNo.NO;
+import static uk.gov.hmcts.reform.fpl.enums.YesNo.YES;
+import static uk.gov.hmcts.reform.fpl.utils.AgeDisplayFormatHelper.formatAgeDisplay;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateToString;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+public class CaseSubmissionTemplateDataGenerationService
+    extends DocmosisTemplateDataGeneration<DocmosisCaseSubmission> {
+    private static final String NEW_LINE = "\n";
+    private static final String SPACE_DELIMITER = " ";
+    private static final String DEFAULT_STRING = "-";
+    private static final String CONFIDENTIAL = "Confidential";
+
+    private final Time time;
+    private final UserDetailsService userDetailsService;
+    private final HmctsCourtLookupConfiguration courtLookupConfiguration;
+
+    public DocmosisCaseSubmission getTemplateData(final CaseData caseData) {
+        DocmosisCaseSubmission.Builder applicationFormBuilder = DocmosisCaseSubmission.builder();
+
+        applicationFormBuilder
+            .applicantOrganisations(getApplicantsOrganisations(caseData.getAllApplicants()))
+            .respondentNames(getRespondentsNames(caseData.getAllRespondents()))
+            .courtName(courtLookupConfiguration.getCourt(caseData.getCaseLocalAuthority()).getName())
+            .submittedDate(formatDateDisplay(time.now().toLocalDate()))
+            .ordersNeeded(getOrdersNeeded(caseData.getOrders()))
+            .directionsNeeded(getDirectionsNeeded(caseData.getOrders()))
+            .allocation(caseData.getAllocationProposal())
+            .hearing(buildDocmosisHearing(caseData.getHearing()))
+            .hearingPreferences(buildDocmosisHearingPreferences(caseData.getHearingPreferences()))
+            .internationalElement(buildDocmosisInternationalElement(caseData.getInternationalElement()))
+            .risks(buildDocmosisRisks(caseData.getRisks()))
+            .factorsParenting(buildDocmosisFactorsParenting(caseData.getFactorsParenting()))
+            .respondents(buildDocmosisRespondents(caseData.getAllRespondents()))
+            .applicants(buildDocmosisApplicants(caseData.getAllApplicants(), caseData.getSolicitor()))
+            .children(buildDocmosisChildren(caseData.getAllChildren()))
+            .others(buildDocmosisOthers(caseData.getAllOthers()))
+            .proceeding(buildDocmosisProceedings(caseData.getAllProceedings()))
+            .relevantProceedings(getValidAnswerOrDefaultValue(caseData.getRelevantProceedings()))
+            .groundsForEPOReason(isNotEmpty(caseData.getOrders())
+                ? getGroundsForEPOReason(caseData.getOrders().getOrderType(),
+                    caseData.getGroundsForEPO())
+                : DEFAULT_STRING)
+            .groundsThresholdReason(buildGroundsThresholdReason(caseData.getGrounds()))
```

Comment:
Could this ternary be moved into the method? Would then keep the clean look of the rest of this method.